Objects/object.c

```diff
@@ -2525,10 +2521,7 @@ _Py_ResurrectReference(PyObject *op)
 #ifdef Py_TRACE_REFS
     _Py_AddToAllObjects(op);
 #endif
-    if (_PyRuntime.ref_tracer.tracer_func != NULL) {
-        void* data = _PyRuntime.ref_tracer.tracer_data;
-        _PyRuntime.ref_tracer.tracer_func(op, PyRefTracer_CREATE, data);
-    }
+    _PyReftracerTrack(op, PyRefTracer_CREATE);
```

Comment:
This seems incorrect.

The tracemalloc hook should trigger when the object is freed, not when its `tp_dealloc` is called.
So this object should still be live, IIRC.

Reply:
This is a creation call not a destruction call (_Py_ResurrectReference). Notice that before this there is a call to `_Py_AddToAllObjects` so this is marking the creation of a new object correctly

Reply:
Ah, but indeed there is no "new object pointer" being created here. That's a good point.